tests/worker/test_activity.py

```diff
@@ -1351,3 +1352,40 @@ def assert_activity_application_error(
     ret = assert_activity_error(err)
     assert isinstance(ret, ApplicationError)
     return ret
+
+
+@activity.defn
+async def activity_with_retry_delay(retry_delay_seconds: float):
+    if activity.info().attempt == 1:
+        raise ApplicationError(
+            "Deliberately failing with next_retry_delay set",
+            next_retry_delay=timedelta(seconds=retry_delay_seconds),
+        )
+
+
+@workflow.defn
+class ActivitiesWithRetryDelayWorkflow:
+    @workflow.run
+    async def run(self, retry_delay_seconds: float) -> float:
+        t0 = workflow.time()
+        await workflow.execute_activity(
+            activity_with_retry_delay,
+            retry_delay_seconds,
+            schedule_to_close_timeout=timedelta(seconds=retry_delay_seconds * 2),
+        )
+        t1 = workflow.time()
+        return t1 - t0
+
+
+async def test_activity_retry_delay(client: Client):
+    retry_delay = timedelta(seconds=2)
```

Comment:
Will this make the test take 2 seconds? I think we can just check whether the property was accepted by the server and set on the way back (e.g. see https://github.com/temporalio/sdk-dotnet/pull/254/files#diff-bfb9d5dbaaa6ec1dfd5ce1837ad44f89fdb2bb75195af55b207ea9627a521eb6) instead of the actual server behavior

Reply:
Yes, I was following what we do in the Typescript test: https://github.com/temporalio/sdk-typescript/blob/main/packages/test/src/test-integration-workflows.ts#L404-L436

Reply:
I think we can just test whether the exception has this property set instead of testing server behavior. These kinds of tests become flaky in slower environments. Here's an example in .NET where we only confirm property is set, not actual server timing: https://github.com/temporalio/sdk-dotnet/blob/4a6d4a0e886360242ea66d3402095b9b4406d690/tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs#L4557. I think we should be able to trust if the server has the property set it will do the right thing.